### DISCUSSION TITLE
Request for improved tagging, name collision behaviour, etc.

### DISCUSSION THREAD

**Discussion by peter-ayla** [audiamus/AaxAudioConverter#59] [Ideas]
First of all, I must say this is an incredible utility.

I've been converting my AAX files using a combination of much more limited, more cumbersome, and slower tools, having to do many things like tagging and chapter-splitting manually. Having one tool that does it all for me in one operation - and much faster - is a godsend. And the documentation is world-class. I love this utility so much that I'm reconverting all my AAX files again with it. Thank you so much for your efforts!

I do have a couple of suggestions for features that I would find useful.
* Being able to select the name collision behaviour before starting a batch of conversions. A few times I've started a batch of 50 files to convert overnight, only to find the next morning that the job stopped on the fifth one because of a name collision. Specifying it as part of the job would allow long jobs to run truly unattended.
* more powerful tagging. For example, I like to include the total tracks in each file tag and name (eg. Hannibal - 1 of 25.mp3), so having a "totaltracks" tag would allow this. Also, having different tags for the long book title and short book title name would be helpful - I usually want to use the long form for the directory but the short form for the file tag and name, and having different tags would allow me to specify that convention for a whole batch.
* an option to set the conversion order by AAX file creation date. I download all my AAX files into a single directory, but convert them in batches. Having the mp3 file creation order be the same as the AAX file creation order makes it easy to see what AAX files have not yet been converted when I start a new batch.

**Comment by audiamus** [comment]
Thank you very much for your suggestions.

>Being able to select the name collision behaviour before starting a batch of conversions.

That should be doable. But it will become optional. Because if enabled, the program will always ask for the behavior to be applied, as it can only actually detect a conflict once it starts converting that book.

> more powerful tagging

Difficult. It's already complex enough with all the possible combinations, as it's affected not only by the setting in the property grid but also by conversion mode, single- or multipart and one or two options in the Basic Settings (preferences). But I will have a look at an option for the total number of tracks. Can't make promises, though.

> an option to set the conversion order by AAX file creation date

Could also be doable, again as an option. Such a date column would not be there by default. But if it's visible it should be sortable more or less automatically.

I'll put these suggestions (the feasible ones) on the to-do list but I am not able to give any time frame.

**Reply by peter-ayla** [reply]
Thank you very much for your consideration of these feature requests.

> > Being able to select the name collision behaviour before starting a batch of conversions.
> 
> That should be doable. But it will become optional. Because if enabled, the program will always ask for the behavior to be applied, as it can only actually detect a conflict once it starts converting that book.

Perhaps having a drop down menu for "File exists action" below the "Mode" section of the main page, with options of "Always ask" (default), "Overwrite", "New file/folder", or "Skip", would allow someone to set the behaviour before starting the conversion? Either there, or something similar in the program settings? The action could then be applied automatically if/when a conflict occurs.

Just a suggestion, of course.

> But I will have a look at an option for the total number of tracks. Can't make promises, though.

That would be great! Thank you!

> 
> > an option to set the conversion order by AAX file creation date
> 
> Could also be doable, again as an option. Such a date column would not be there by default. But if it's visible it should be sortable more or less automatically.

If the list of books could be sorted using a Created Date or Modified Date column, and the conversion process respected that ordering, that would be perfect.

(Right now, the conversion process seems to convert the books in alphabetical title order, no matter how they are sorted visually in the list)

Thanks again for your consideration. I should emphasize that your utility is already 100 times better than any others I've used for this process, so please don't interpret my suggestions as complaints! It's so satisfying to use a tool that does what it's designed to do so efficiently, and so well.

**Reply by audiamus** [reply]
> Perhaps having a drop down menu for "File exists action" below the "Mode" section of the main page, with options of "Always ask" (default), "Overwrite", "New file/folder", or "Skip", would allow someone to set the behaviour before starting the conversion? Either there, or something similar in the program settings? The action could then be applied automatically if/when a conflict occurs.

Yes, it should become an option with the default value "ask" and the other values as in the existing message box. It shall go to the settings dialog.

Regarding the long and short book title, something might be possible with the additional meta data in the SimsBySeries folder (Audible App file structure), see #62, but it would only be usable for books downloaded with the Audible App, as we need a key for those files and even the ASIN may not suffice. That key appears to be the SKU.
